src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -0,0 +1,45 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.UserDto;
+import com.hillel.items_exchange.model.BaseEntity;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/user")
+@CrossOrigin(origins = "http://localhost:4200")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final UserService userService;
+
+    @GetMapping("/info/{id}")
+    public @ResponseBody
+    ResponseEntity<UserDto> getUserInfo(@PathVariable("id") long id, Principal principal) {
+
+        userService.findByUsernameOrEmail(principal.getName())
+                .map(BaseEntity::getId)
+                .filter(userId -> userId == id)
+                .orElseThrow(() -> new AccessDeniedException("You do not have access to view the data of another user!"));
+
+        return userService.getUserDto(id)
+                .map(userDto -> new ResponseEntity<>(userDto, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
```

Comment:
I think, that it would be more appropriate here to throw an exception with detailed information

Reply:
Can I give you more details? How do you imagine that?

Reply:
@21November I propose you to change what you've done here in such way:
1. To create additional method into the service layer ``Optional<UserDto> getByUsernameOrEmail`` that will return you DTO. Under the hood you will use userRepository.findByUsername(usernameOrEmail) to get those information.
2. After getting your UserDto you should compare it's id with that that you received into ``getUserInfo(@PathVariable("id") long id``.
3. Return that user if id equals and return AccessDenied exception with description or 403 (Forbidden) code.
Doing that way you decrease quantity of requests to the database and make the code simpler.